statsmodels/genmod/families/family.py

```diff
@@ -72,7 +75,14 @@ def _getlink(self):
     link = property(_getlink, _setlink, doc="Link function for family")
 
     def __init__(self, link, variance):
-        self.link = link()
+        if inspect.isclass(link):
```

Comment:
I think we can avoid loading the inspect module and check directly
`isinstance(link, type)`